src/vos/vos_io.c

```diff
@@ -1599,22 +1632,38 @@ vos_publish_blocks(struct vos_container *cont, d_list_t *blk_list, bool publish,
 	hint_ctxt = cont->vc_hint_ctxt[ios];
 	D_ASSERT(hint_ctxt);
 
-	rc = publish ? vea_tx_publish(vsi, hint_ctxt, blk_list) :
-		       vea_cancel(vsi, hint_ctxt, blk_list);
-	if (rc)
+	if (publish) {
+		if (dth != NULL) {
+			struct dtx_rsrvd_uint	*dru;
+
+			D_ASSERT(dth->dth_op_seq >= 1);
+
+			dru = &dth->dth_rsrvds[dth->dth_op_seq - 1];
+			D_ASSERT(dru->dru_scm_ptr != NULL);
```

Comment:
Because we need to initialize 'dru_nvme_list', that is done during vos_publish_scm(), since we always call vos_publish_scm() firstly, so I add such initialization in vos_publish_scm().